JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/movies/MovieDetailsScreen.kt

```diff
@@ -52,26 +54,48 @@ object MovieDetailsScreen {
 
 @Composable
 fun MovieDetailsScreen(
-    movieId: String,
     goToMoviePlayer: () -> Unit,
     onBackPressed: () -> Unit,
     refreshScreenWithNewMovie: (Movie) -> Unit
 ) {
-    val childPadding = rememberChildPadding()
-    val screenHeight = LocalConfiguration.current.screenHeightDp.dp
-    val movieRepository = LocalMovieRepository.current!!
-    val movieDetails = remember { movieRepository.getMovieDetails(movieId = movieId) }
+    val movieDetailsScreenViewModel: MovieDetailsScreenViewModel =
+        viewModel(factory = MovieDetailsScreenViewModel.Factory)
+
+    val loadingState by movieDetailsScreenViewModel.loadingState.collectAsState()
+
+    when (loadingState) {
+        is LoadingState.Loading -> {
+            Text(text = "Loading...")
+        }
 
-    LaunchedEffect(Unit) {
-        movieRepository.getMovieDetails(movieId = movieId)
+        is LoadingState.Done -> {
+            Details(
+                movieDetails = (loadingState as LoadingState.Done).movieDetails,
```

Comment:
It seems to me that this cast is redundant. The when expression implies that the  `loadingState`  is already of type  `LoadingState.Done`